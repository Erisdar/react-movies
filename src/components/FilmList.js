import React from 'react';
import FilmItem from './film/FilmItem';
import NoFilms from './NoFilms';

export default ({ films, changeView }) => (
    <div className="film-list align-self-center d-flex flex-row justify-content-around flex-wrap">
        {
            films.length > 0
                ? <>
                    {films.map((film) => <FilmItem film={film} key={film.id} moreDetail={() => changeView(film)} />)}
                </>
                : <NoFilms />
        }
    </div>
);