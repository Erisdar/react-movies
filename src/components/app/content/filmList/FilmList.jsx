import React from 'react';
import FilmItem from './filmItem/FilmItem';
import './filmList.css';
import NoFilms from './noFilms/NoFilms';


export default ({ films, changeView }) => (
    <div className="film-list content__film-list align-self-center d-flex flex-row justify-content-around flex-wrap">
        {
            films.length > 0
                ? <>
                    {films.map((film) => <FilmItem film={film} key={film.id} moreDetail={() => changeView(film)} />)}
                </>
                : <NoFilms />
        }
    </div>
);