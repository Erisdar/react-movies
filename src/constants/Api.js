export const LOAD_FILMS = "https://reactjs-cdp.herokuapp.com/movies";