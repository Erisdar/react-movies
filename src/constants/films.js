export const POPULAR = "popular";
export const FOUNDED = "founded";