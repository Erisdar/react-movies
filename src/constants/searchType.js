export const TITLE = 'title';
export const GENRES = 'genres';