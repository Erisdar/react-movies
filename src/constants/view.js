export const BRIEFLY = 'briefly';
export const DETAIL = 'detail';