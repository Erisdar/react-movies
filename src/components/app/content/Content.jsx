import ErrorBoundary from 'error/ErrorBoundary';
import React, { Component } from 'react';
import { filterFilm } from 'util/filmUtil';
import DetailFilmInfo from './detailFilmInfo/DetailFilmInfo';
import FilmList from './filmList/FilmList';
import Search from './search/Search';

function getFilms() {
    return require('data/film.json');
};

export default class Content extends Component {

    constructor(props) {
        super(props);
        this.state = {
            films: [],
            searchType: 'title',
            searchValue: '',
            view: 'briefly',
            detailFilm: null
        };
    }

    setSearch(searchType, searchValue) {
        this.setState({
            searchType: searchType,
            searchValue: searchValue
        })
    };

    setView(view, detailFilm) {
        this.setState({
            view: view,
            detailFilm: detailFilm
        })
    }

    componentDidMount() {
        this.setState({ films: getFilms() })
    };

    render() {
        let filteredFilms = filterFilm(this.state.films, this.state.searchType, this.state.searchValue);

        return (
            <div className="content d-flex flex-column mb-auto">
                {
                    this.state.view === 'briefly'
                        ? <Search onSearch={this.setSearch.bind(this)} />
                        : <DetailFilmInfo changeView={() => { this.setView('briefly', null); this.setSearch('title', ''); }} film={this.state.detailFilm} />
                }
                <ErrorBoundary>
                    <FilmList films={filteredFilms} changeView={(film) => { this.setView('detail', film); this.setSearch('genre', film.genre); }} />
                </ErrorBoundary>
            </div>
        );
    }
};