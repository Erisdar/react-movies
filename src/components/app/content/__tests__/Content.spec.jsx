import { BRIEFLY, DETAIL } from 'constants/view';
import { mount } from 'enzyme';
import React from 'react';
import Content from '../Content';
import FilmList from '../filmList/FilmListContainer';

jest.mock('../search/SearchContainer', () => {
    return jest.fn().mockImplementation(() => {
        return (<div className='content__search'></div>)
    });
});

jest.mock('../filmList/FilmListContainer', () => {
    return jest.fn().mockImplementation(() => {
        return (<div className='content__film-list'></div>)
    });
});

jest.mock('../detailFilmInfo/DetailFilmInfoContainer', () => {
    return jest.fn().mockImplementation(() => {
        return (<div className='content__detail-film-info'></div>)
    });
});

const films = require('data/film.json').filter(film => film.id == 1);

describe('Content component', () => {

    it('should be render correctly with SEARCH', () => {
        let foundFilms = jest.fn();
        const component = mount(
            <Content view={BRIEFLY} popular={[]} loadPopularFilms={foundFilms} />
        );
        expect(foundFilms).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });

    it('should be render correctly with DETAIL INFO', () => {
        let foundFilms = jest.fn();
        const component = mount(<Content view={DETAIL} popular={[]} loadPopularFilms={foundFilms} />);
        expect(foundFilms).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });

    it('should be render correctly with ErrorBoundary', () => {
        console.error = jest.fn();
        console.log = jest.fn();
        let foundFilms = jest.fn();
        FilmList.mockImplementationOnce(() => {
            throw new Error('New Error');
        });
        const component = mount(<Content view={BRIEFLY} popular={[]} loadPopularFilms={foundFilms} />);
        expect(foundFilms).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });

    it('should not be call loadPopularFilms()', () => {
        let foundFilms = jest.fn();
        mount(<Content view={DETAIL} popular={films} loadPopularFilms={foundFilms} />);
        expect(foundFilms).not.toHaveBeenCalled();
    });
});