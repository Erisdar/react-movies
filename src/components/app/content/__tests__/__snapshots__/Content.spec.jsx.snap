// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should be render correctly with DETAIL INFO 1`] = `
<Content
  loadPopularFilms={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  popular={Array []}
  view="detail"
>
  <div
    className="content d-flex flex-column mb-auto"
  >
    <mockConstructor>
      <div
        className="content__detail-film-info"
      />
    </mockConstructor>
    <ErrorBoundary>
      <mockConstructor>
        <div
          className="content__film-list"
        />
      </mockConstructor>
    </ErrorBoundary>
  </div>
</Content>
`;

exports[`Content component should be render correctly with ErrorBoundary 1`] = `
<Content
  loadPopularFilms={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  popular={Array []}
  view="briefly"
>
  <div
    className="content d-flex flex-column mb-auto"
  >
    <mockConstructor>
      <div
        className="content__search"
      />
    </mockConstructor>
    <ErrorBoundary>
      <div
        className="film-list content__film-list d-flex align-self-center flex-column justify-content-start"
      >
        <h1>
          Sorry, server is not responding now
        </h1>
      </div>
    </ErrorBoundary>
  </div>
</Content>
`;

exports[`Content component should be render correctly with SEARCH 1`] = `
<Content
  loadPopularFilms={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  popular={Array []}
  view="briefly"
>
  <div
    className="content d-flex flex-column mb-auto"
  >
    <mockConstructor>
      <div
        className="content__search"
      />
    </mockConstructor>
    <ErrorBoundary>
      <mockConstructor>
        <div
          className="content__film-list"
        />
      </mockConstructor>
    </ErrorBoundary>
  </div>
</Content>
`;
